Description: default-config improvements.
  Fix a bug when starting fvwm that resets the chosen background image.
.
  Add $FVWM_USERDIR/local.config as a local override to the default-config,
  to allow local modifications to the config.
Author: Jaimos Skriletz <jaimosskriletz@gmail.com>
Forwarded: not-needed.
Last-Update: 2022-10-09
--- a/default-config/config
+++ b/default-config/config
@@ -55,14 +55,19 @@ InfoStoreAdd terminal x-terminal-emulato
 # actions that are run after a restart.
 DestroyFunc StartFunction
 AddToFunc   StartFunction
-+ I Test (Init, f $[FVWM_USERDIR]/.BGdefault) \
-    Exec exec fvwm2-root $[FVWM_USERDIR]/.BGdefault
-+ I TestRc (NoMatch) Exec exec fvwm2-root \
-    $[FVWM_DATADIR]/default-config/images/background/bg1.png
++ I Test (Init) InitBackground
 + I Test (Init) Module FvwmBanner
 + I Module FvwmButtons RightPanel
 + I Module FvwmEvent EventNewDesk

+# Function to set background when fvwm starts
+DestroyFunc InitBackground
+AddToFunc InitBackground
++ I Test (f $[FVWM_USERDIR]/.BGdefault) \
+    Exec exec fvwm2-root $[FVWM_USERDIR]/.BGdefault
++ I TestRc (NoMatch) Exec exec fvwm2-root \
+    $[FVWM_DATADIR]/default-config/images/background/bg1.png
+
 # Mouse Bindings Functions
 DestroyFunc RaiseMoveX
 AddToFunc RaiseMoveX
@@ -654,3 +659,11 @@ DestroyModuleConfig EventNewDesk:*
 # This sets the default colorsets.
 *FvwmFormDefault: Colorset 10
 *FvwmFormDefault: ItemColorset 13
+
+# Local configuration file. For use with the default-config.
+#
+# If $FVWMUSER_DIR/local.config ($HOME/.fvwm/local.config by default)
+# exists, then read it. This allows changes to default-config settings
+# without needing a full copy of the default-config. This will also allow
+# default-config changes to get used after upgrades (if applicable).
+Test (f $[FVWM_USERDIR]/local.config) Read $[FVWM_USERDIR]/local.config
